src/data/data.cc

```diff
@@ -698,6 +698,9 @@ void MetaInfo::Extend(MetaInfo const& that, bool accumulate_rows, bool check_col
     this->feature_type_names = that.feature_type_names;
     auto &h_feature_types = feature_types.HostVector();
     LoadFeatureType(this->feature_type_names, &h_feature_types);
+  } else if (!that.feature_types.Empty()) {
```

Comment:
@rongou Could you please help comment on when the meta info has feature types but doesn't have type names?

Reply:
I think in some tests we set the feature types directly without setting the type names.

Reply:
Got it, let me try to run the tests.